spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresChannelMessageTableSubscriber.java

```diff
@@ -0,0 +1,161 @@
+package org.springframework.integration.jdbc.channel;
+
+import org.postgresql.PGNotification;
+import org.postgresql.jdbc.PgConnection;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.core.log.LogAccessor;
+import org.springframework.integration.jdbc.store.JdbcChannelMessageStore;
+import org.springframework.integration.util.UUIDConverter;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+import java.sql.Statement;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class PostgresChannelMessageTableSubscriber implements SmartLifecycle {
+
+	private final LogAccessor logger = new LogAccessor(this.getClass()); // NOSONAR final
```

Comment:
We don't need `NOSONAR` here since the property is `private`.
We use it in those case where it is `protected`.
Also this one can simply be `static`: I doubt any one is going to extend this class.

Reply:
Indeed, I fixed that.

Reply:
I made the class final to avoid its subclassing as I agree that there's no point in it.